kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerRuleCapabilitiesIntegrationTest.java

```diff
@@ -0,0 +1,174 @@
+                                                    KieContainerStatus.STARTED,
+                                                    null);
+        try {
+            controllerClient.scanNow(container);
+            fail("Should throw exception about container not found.");
+        } catch (KieServerControllerClientException e) {
+            assertNotFoundException((T) e);
+        }
+    }
+
+    @Test
+    public void testStartAndStopScanner() throws Exception {
+        ServerTemplate serverTemplate = createServerTemplate();
+        ContainerSpec container = startContainerWithVersion(RELEASE_ID, serverTemplate);
+
+        checkKieContainerResource(RELEASE_ID, RELEASE_ID);
+
+        controllerClient.startScanner(container, 5L);
+        controllerClient.stopScanner(container);
+
+        checkKieContainerResource(RELEASE_ID, RELEASE_ID);
+    }
+
+    @Test
+    public void testStartScannerNotExistingContainer() {
+        ServerTemplate serverTemplate = createServerTemplate();
+        ContainerSpec container = new ContainerSpec("not-existing",
+                                                    "not-existing",
+                                                    serverTemplate,
+                                                    RELEASE_ID,
+                                                    KieContainerStatus.STARTED,
+                                                    null);
+
+        try {
+            controllerClient.startScanner(container,
+                                          1000L);
+            fail("Should throw exception about container not found.");
+        } catch (KieServerControllerClientException e) {
+            assertNotFoundException((T) e);
+        }
+    }
+
+    @Test
+    public void testStopScannerNotExistingContainer() {
+        ServerTemplate serverTemplate = createServerTemplate();
+        ContainerSpec container = new ContainerSpec("not-existing",
+                                                    "not-existing",
+                                                    serverTemplate,
+                                                    RELEASE_ID,
+                                                    KieContainerStatus.STARTED,
+                                                    null);
+        try {
+            controllerClient.stopScanner(container);
+            fail("Should throw exception about container not found.");
+        } catch (KieServerControllerClientException e) {
+            assertNotFoundException((T) e);
+        }
+    }
+
+    @Test
+    public void testStopNotRunningScanner() throws Exception {
+        ServerTemplate serverTemplate = createServerTemplate();
+        ContainerSpec container = startContainerWithVersion(RELEASE_ID,
+                                                            serverTemplate);
+
+        checkKieContainerResource(RELEASE_ID,
+                                  RELEASE_ID);
+
+        controllerClient.stopContainer(container);
```

Comment:
@jakubschwan in this one specifically I don't think we should throw an exception in case the container is already stopped. @mswiderski wdyt?

Reply:
Makes sense, I agree too